bootstrap/kubeadm/api/v1alpha4/kubeadmconfig_types.go

```diff
@@ -94,6 +97,34 @@ type KubeadmConfigSpec struct {
 	// For more information, refer to https://github.com/kubernetes-sigs/cluster-api/pull/2763#discussion_r397306055.
 	// +optional
 	UseExperimentalRetryJoin bool `json:"useExperimentalRetryJoin,omitempty"`
+
+	// Ignition contains Ignition specific configuration.
+	//
+	// If set, bootstrap data will be generated in Ignition format instead of cloud-init.
+	//
+	// +optional
+	Ignition *IgnitionSpec `json:"ignitionConfig,omitempty"`
```

Comment:
```suggestion
	AdditionalIgnitionSpec *IgnitionSpec `json:"ignitionConfig,omitempty"`
```
Or
```
   UserIgnitionSpec *IgnitionSpec `json:"ignitionConfig,omitempty"`
```

Reply:
@vincepri @randomvariable @MarcelMue before I start changing things back and forth, what do you think? :)

Reply:
Happy to keep using current name, per https://github.com/kubernetes-sigs/cluster-api/pull/4172#discussion_r612141838

Reply:
I'll mark as resolved.